src/main/java/org/candlepin/insights/inventory/InventoryService.java

```diff
@@ -117,9 +118,17 @@ protected CreateHostIn createHost(ConduitFacts facts, OffsetDateTime syncTimesta
         host.setMacAddresses(facts.getMacAddresses());
         host.setInsightsId(facts.getInsightsId());
 
+        host.setSystemProfile(createSystemProfile(facts));
```

Comment:
@kahowell have you verified that inventory will not reset all fields for this system profile by doing this? In particular, createSystemProfile will return a new SystemProfileIn object instead of null. I would assume that null means no update, but not sure about a defaulted object instance. Might consider having createSystemProfile return null if there is nothing to set. I'm assuming that it is all Ok, but wanted to check with you to verify we understood how inventory behaves in this case.

Reply:
Great point. I'll look at the generated JSON and see how HBI behaves with it in a dev environment.

Reply:
I verified that the generated JSON does contain an empty object for system profile if we can't determine cloud provider. However, in my testing, HBI does not treat this as "unset all system profile fields". It treats it as "no update to system profile". (I verified by launching a local instance of HBI, and manually adding a few fields to system profile including those that were in arrays, and then having a modified stub pinhead update that host). Also, when cloud provider is already set, it's not unset.

In the case that cloud provider is set in our JSON, but some other field already exists on the host record in HBI, HBI happily touches *just* cloud provider.

FWIW, this is what makes that work for us, thanks to your earlier work :smile: : https://github.com/RedHatInsights/rhsm-conduit/blob/0bc7fd090bed447d97be8e76a35b0a7755164bdb/src/main/java/org/candlepin/insights/jackson/ObjectMapperContextResolver.java#L58-L59